include/zephyr/acpi/acpi.h

```diff
@@ -180,4 +276,15 @@ int acpi_drhd_get(enum AcpiDmarScopeType scope, ACPI_DMAR_DEVICE_SCOPE *dev_scop
  * @return local apic info on success or NULL otherwise
  */
 ACPI_MADT_LOCAL_APIC *acpi_local_apic_get(int cpu_num);
+
+/**
+ * @brief invoke an acpi method and return the result.
+ *
+ * @param path the path name of the acpi object
+ * @param arg_list the list of arguments to be pass down
+ * @param ret_obj the acpi result to be return
+ * @return return 0 on success or error code
+ */
+int acpi_invoke_method(char *path, ACPI_OBJECT_LIST *arg_list, ACPI_OBJECT *ret_obj);
```

Comment:
Could it be `const char *path`? Or ACPICA API is missing const so you'd have to typecast internally in that case?